src/basic/blockdev-util.c

```diff
@@ -177,3 +177,24 @@ int get_block_device_harder(const char *path, dev_t *ret) {
 
         return 1;
 }
+
+int get_root_or_usr_block_dev(dev_t *ret_devno) {
+        int r;
+        dev_t devno;
+
+        r = get_block_device_harder("/", &devno);
+        if (r < 0)
+                return log_error_errno(r, "Failed to determine block device of root file system: %m");
```

Comment:
if this is now a library function this should probably be downgraded to debug logging